openwisp_utils/admin_theme/menu.py

```diff
@@ -0,0 +1,154 @@
+from django.apps import registry
+from django.core.exceptions import ImproperlyConfigured
+from django.urls import reverse
+from django.utils.translation import gettext as _
+
+from ..utils import SortedOrderedDict
+
+MENU = SortedOrderedDict()
+
+
+class BaseMenuItem:
+    def __init__(self, config):
+        if not isinstance(config, dict):
+            raise ImproperlyConfigured(
+                f'"config" should be a type of "dict". Error for config-{config}'
+            )
+
+    def get_context(self, request):
+        return self.create_context(request)
+
+    def create_context(self, request):
+        return {'label': self.label, 'url': self.url, 'icon': self.icon}
+
+
+class MenuItem(BaseMenuItem):
+    def __init__(self, config):
+        super().__init__(config)
+        if config.get('name'):
+            if not isinstance(config['name'], str):
+                raise ValueError(
+                    f'"name" of menu item should be a type of "str". Error for config-{config}'
+                )
+            self.name = config['name']
+        else:
+            raise ValueError(f'"name" is missing in the config-{config}')
+        if not config.get('model'):
+            raise ValueError(f'"model" is missing in config-{config}')
+        self.model = config['model']
+        self.label = _(self.get_label(config))
+        self.icon = config.get('icon')
+
+    def get_label(self, config=None):
+        if config:
+            if config.get('label'):
+                return config['label']
+            app_label, model = config['model'].split('.')
+            model_class = registry.apps.get_model(app_label, model)
+            return f'{model_class._meta.verbose_name_plural} {self.name}'
+        return self.label
```

Comment:
why are you returning `self.label` in the end while returning `config['label']` above?

Reply:
I think it was a mistake. Fixed it. :+1: